x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/cache/CacheFile.java

```diff
@@ -112,20 +112,20 @@ public boolean acquire(final EvictionListener listener) throws IOException {
         ensureOpen();
         boolean success = false;
         if (refCounter.tryIncRef()) {
-            try (ReleasableLock ignored = evictionLock.acquire()) {
-                try {
-                    ensureOpen();
-                    final Set<EvictionListener> newListeners = new HashSet<>(listeners);
-                    final boolean added = newListeners.add(listener);
-                    assert added : "listener already exists " + listener;
-                    maybeOpenFileChannel(newListeners);
-                    listeners = Collections.unmodifiableSet(newListeners);
-                    success = true;
-                } finally {
-                    if (success == false) {
-                        refCounter.decRef();
-                    }
+            evictionLock.lock();
+            try {
+                ensureOpen();
+                final Set<EvictionListener> newListeners = new HashSet<>(listeners);
+                final boolean added = newListeners.add(listener);
+                assert added : "listener already exists " + listener;
+                maybeOpenFileChannel(newListeners);
+                listeners = Collections.unmodifiableSet(newListeners);
+                success = true;
+            } finally {
+                if (success == false) {
+                    refCounter.decRef();
                 }
+                evictionLock.unlock();
```

Comment:
Looks like `refCounter.decRef()` can throw an `UncheckedIOException` so we need another `finally` here.

Reply:
Good catch (bis)

Reply:
I pushed a0e6db7e6a794f38d3da6b85138785d153ebe7a2